restapi.go

```diff
@@ -224,7 +225,8 @@ func (self *RestAPI) pushNotification(reqId string, kv map[string]string, perdp
 	notif := NewEmptyNotification()
 
 	for k, v := range kv {
-		if len(v) <= 0 {
+		v = strings.Trim(v, " \n")
```

Comment:
The length of the string can never be less than zero and a zero length
string is not a valid value.

I added the trim to fix things like param=" test ". A param with leading
our trailing white space is not a valid uniqush param.

If it's in a parameter intended to be passed in a payload, it'll just take
up extra space.

I made the generalization that a parameter whose value was a string with
whitespace only was a mistake. This follows the same logic as zero length
string. PushType ="" and pushType = " ", may both be valid in a payload,
but they probably aren't.

Tina
On Jun 9, 2015 9:30 PM, "Misha Nasledov" <notifications@github.com> wrote:

> In restapi.go
> <https://github.com/uniqush/uniqush-push/pull/66#discussion_r32087414>:
>
> > @@ -224,7 +225,8 @@ func (self *RestAPI) pushNotification(reqId string, kv map[string]string, perdp
> >  	notif := NewEmptyNotification()
> >
> >  	for k, v := range kv {
> > -		if len(v) <= 0 {
> > +		v = strings.Trim(v, " \n")
>
> I'm curious why we had to add the Trim here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uniqush/uniqush-push/pull/66/files#r32087414>.
>